tests/test_controller.py

```diff
@@ -553,8 +554,9 @@ def test_borrow_and_fulfill_with_streaming_delivery_mechanism(self):
                     datetime.datetime.utcnow() + datetime.timedelta(seconds=3600),
                 )
             )
-            response = self.manager.loans.borrow(
-                data_source.name, identifier.type, identifier.identifier)
+            with self.temp_config():
```

Comment:
There are a number of places where I had to use `self.temp_config()` to create a configuration object where I didn't before. I think this is because of the change I mentioned earlier, on line 288 of `controller.py`. That code makes sure the AuthdataUtility is fully configured if the `library_registry_shared_secret` is set in the database. This always happens, since we call `MockAdobeConfiguration.initialize_library` at the start of these tests when setting up the `AuthdataUtility`. The configuration we create to set up the `AuthdataUtility` goes out of scope, but the change we made to the database persists throughout the test.

The rule is that the AuthdataUtility doesn't have to be configured if the configuration was completely absent. Most of the time, that's what used to happen. We would get to `CirculationManagerAnnotator._adobe_patron_identifier`, try to look up AuthdataUtility, notice that there is no configuration (since we're not testing that part of the code), and that part of the code would be skipped. But now, the config goes out of scope and the database change doesn't. We try to look up AuthdataUtility and get a crash, because the database part of the configuration is set up but the config part is not.